presto-spi/src/main/java/com/facebook/presto/spi/security/AuthorizedIdentity.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.security;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class AuthorizedIdentity
+{
+    private final String userName;
+    private final Optional<String> reasonForSelect;
```

Comment:
Is this supposed to capture error message when the user failed authorization. If so, should this be renamed to failureReason

Reply:
This is not to capture error message. It is to tell why the username is selected from a bunch of identities from the clients. If there is not valid username, the request will fail directly without returning the authorized identity (at least this is the goal to achieve in DIPS side).